File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.size();i++){
            if(s[i]==')' || s[i]=='}' || s[i]==']'){
                if(!st.empty() && ((s[i]==')' && st.top()=='(') || 
                (s[i]=='}' && st.top()=='{')||
                (s[i]==']' && st.top()=='['))){
                    st.pop();
                }
                else{
                    return false;
                }
            }
             
            else{
                st.push(s[i]);
            }
        }
        if(!st.empty())return false;
        return true;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        int n = s.length();
        Deque<Character> st = new ArrayDeque<>();
        for(int i = 0;i<n;i++){
            if(s.charAt(i)=='(' || s.charAt(i) == '{' || s.charAt(i) == '['){
                st.push(s.charAt(i));
            }
            else{
                if(st.isEmpty()){
                    return false;
                }
                if(s.charAt(i)==')'){
                    if(st.peek() != '('){
                        return false;
                    }
                    st.pop();
                }
                else if(s.charAt(i)=='}'){
                    if(st.peek() != '{'){
                        return false;
                    }
                    st.pop();
                }
                else{
                    if(st.peek() != '['){
                        return false;
                    }
                    st.pop();
                }
            }
        }
        if(!st.isEmpty()){
            return false;
        }
        return true;
    }
}